packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { useCallback, SetStateAction, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { transparentize } from 'polished';
+import facepaint from 'facepaint';
+import Portal from '@leafygreen-ui/portal';
+import Icon, { Size } from '@leafygreen-ui/icon';
+import { useEventListener } from '@leafygreen-ui/hooks';
+import { uiColors } from '@leafygreen-ui/palette';
+import { css, cx, keyframes } from '@leafygreen-ui/emotion';
+
+export const ModalSize = {
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+} as const;
+
+export type ModalSize = typeof ModalSize[keyof typeof ModalSize];
+
+// breakpoints for different screen sizes
+const small = '767px'; // mobile screens, from 0px - 767px
```

Comment:
Are these going to be global breakpoints we'll be using in other libraries? Should we extract to `lib` or some other package?

Ditto with the facepaint instance?

Reply:
I think it might require a little more discussion? I ported these specific breakpoints from MMS, but can move to lib or somewhere else if we feel that it belongs there now? Let me know what you think